src/TraceEvent/Symbols/AdalAuth.cs

```diff
@@ -0,0 +1,59 @@
+﻿#if NET45
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+
+namespace Microsoft.Diagnostics.Symbols
+{
+    public static class Adal
+    {
+        private static object s_lock = new object();
+        private static DateTime s_lastAttempt = DateTime.MinValue;
+        private static string s_accessToken;
+
+        public static string AcquireToken()
+        {
+            lock (s_lock)
+            {
+                if (s_accessToken != null)
+                {
+                    return s_accessToken;
+                }
+
+                // Do not prompt more than once/minute
+                var now = DateTime.UtcNow;
+                if (now < s_lastAttempt + TimeSpan.FromSeconds(60))
+                {
+                    return null;
+                }
+                s_lastAttempt = now;
+            }
+
+            var tenantId = "microsoft.com";
+            var authority = $"https://login.microsoftonline.com/{tenantId}";
+            var resourceId = "499b84ac-1321-427f-aa17-267ca6975798"; // ADO
+
+            var clientId = "1950a258-227b-4e31-a9cf-717495945fc2"; // Azure PowerShell
```

Comment:
Is there something that we need to do to register a clientId for PerfView, and then plumb that through.  I'm a bit weary of merging this as is, since it means we look like Azure PowerShell.

Reply:
You can certainly register PerfView as a 1st-party client application (because you're in Microsoft). Note that the client itself is _not_ secret ("public client" in AAD terms), so that 1st-"partiness" is really about having the app available to all AAD tenants, without needing registration to a particular tenant.

(I don't really know why they built their system like that -- anybody can copy the Azure PowerShell app ID, or any other app ID for that matter, and it really provides 0 security. Oh well.)

Reply:
I would like to have a TraceEvent ID.  If you can point me to documentation internally, I will set this up.

Reply:
You can contact the 1st Party Engagement Team at firpet@microsoft.com

The process is to go to [https://firstpartyportal.msidentity.com](https://firstpartyportal.msidentity.com) (from SAW), and start “Create Application” flow.